core/src/common/cdk-deployer.ts

```diff
@@ -122,143 +158,156 @@ export class CdkDeployer extends cdk.Stack {
+            actions: ['sts:AssumeRole'],
+          }),
+          new PolicyStatement({
+            resources: [`arn:aws:ssm:${Aws.REGION}:${Aws.ACCOUNT_ID}:parameter/cdk-bootstrap/*/*`],
+            actions: ['ssm:PutParameter', 'ssm:GetParameters'],
+          }),
+          new PolicyStatement({
+            resources: [
+              `arn:aws:ecr:${Aws.REGION}:${Aws.ACCOUNT_ID}:repository/cdk*`,
+            ],
+            actions: [
+              'ecr:SetRepositoryPolicy',
+              'ecr:GetLifecyclePolicy',
+              'ecr:PutImageTagMutability',
+              'ecr:DescribeRepositories',
+              'ecr:ListTagsForResource',
+              'ecr:PutImageScanningConfiguration',
+              'ecr:CreateRepository',
+              'ecr:PutLifecyclePolicy',
+              'ecr:SetRepositoryPolicy',
+              'ecr:DeleteRepository',
+              'ecr:TagResource',
+            ],
+          }),
+          new PolicyStatement({
+            resources: [
+              `arn:aws:iam::${Aws.ACCOUNT_ID}:role/cdk*`,
+            ],
+            actions: [
+              'iam:GetRole',
+              'iam:CreateRole',
+              'iam:TagRole',
+              'iam:DeleteRole',
+              'iam:AttachRolePolicy',
+              'iam:DetachRolePolicy',
+              'iam:GetRolePolicy',
+              'iam:PutRolePolicy',
+              'iam:DeleteRolePolicy',
+            ],
+          }),
         new PolicyStatement({
           resources: [
             `arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/codebuild/*`,
-          ],
+        ],
           actions: [
             'logs:PutLogEvents',
           ],
         }),
       ]
     }))
 
-    const codeBuildProject = new Project(this, 'CodeBuildProject', {
-      source: Source.gitHub({
-        owner: props.githubRepository.split('/')[0],
-        repo: props.githubRepository.split('/')[1],
+
+    let source: Source;
+
+    if(props.deploymentType === DeploymentType.WORKSHOP_STUDIO) {
+      const cdkAppSourceCodeBucketName = new CfnParameter(this, 'CDKAppSourceCodeBucketName', {
+        type: 'String',
+      });
+  
+      const cdkAppSourceCodeBucketPrefix = new CfnParameter(this, 'CDKAppSourceCodeBucketPrefix', {
+        type: 'String',
+      });
+
+      source = Source.s3({
+        bucket: Bucket.fromBucketName(
+          this,
+          'CdkAppBucket',
+          cdkAppSourceCodeBucketName.valueAsString
+        ),
+        path: `${cdkAppSourceCodeBucketPrefix.valueAsString}cdk_app.zip`,
+      });
+    } else {
+      source = Source.gitHub({
+        owner: props.githubRepository!.split('/')[0],
+        repo: props.githubRepository!.split('/')[1],
         branchOrRef: props.gitBranch ? props.gitBranch : undefined,
         reportBuildStatus: true,
-      }),
+      });
+    }
+
+
+    const codeBuildProject = new Project(this, 'CodeBuildProject', {
+      source,
       buildSpec: BuildSpec.fromObject({
```

Comment:
might worth to be completely removed to avoid confusion knowing that it is anyway overwritten by the startBuild ...